compiling failed on ppc platform

Hi,

9d6f7f1a32633d6372960db4c16ea13849487257 utils/atomics: added support of GCC atomics intrinsics

This commit introduced a regression for ppc platform. It seems "__sync_add_and_fetch" built-in function is unavailable for ppc platform. Compiling works for ppc64 and ppc64le.

<snip of config.log>
checking compiler support for c11 atomics... no
checking compiler support for built-in atomics... yes  <======
checking valgrind/memcheck.h usability... yes
<snip>

<snip of build log>
1096 libtool: link: gcc -std=gnu99 -Wall -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-bu     ffer-size=4 -grecord-gcc-switches -m32 -mcpu=power7 -mtune=power7 -Wl,-z -Wl,relro -o util/.libs/fi_info util/info.o  src/.libs/lib     fabric.so -lrt -lpthread -ldl
1097 src/.libs/libfabric.so: undefined reference to `__sync_add_and_fetch_8'
1098 collect2: error: ld returned 1 exit status
1099 make[1]: *** [util/fi_strerror] Error 1
1100 make[1]: *** Waiting for unfinished jobs....
1101 src/.libs/libfabric.so: undefined reference to `__sync_add_and_fetch_8'
1102 collect2: error: ld returned 1 exit status
1103 make[1]: *** [util/fi_info] Error 1
1104 src/.libs/libfabric.so: undefined reference to `__sync_add_and_fetch_8'
1105 collect2: error: ld returned 1 exit status
1106 make[1]: *** [util/fi_pingpong] Error 1
1107 make[1]: Leaving directory `/builddir/build/BUILD/libfabric-1.5.1'
1108 make: *** [all] Error 2
1109 error: Bad exit status from /var/tmp/rpm-tmp.YJH3AO (%build)
1110 RPM build errors:
1111     Bad exit status from /var/tmp/rpm-tmp.YJH3AO (%build)
1112 Child returncode was: 1
<snip>

--- a/configure.ac
+++ b/configure.ac
@@ -165,7 +165,11 @@ AC_TRY_LINK([#include <stdint.h>],
     [int32_t a;
      __sync_add_and_fetch(&a, 0);
      __sync_sub_and_fetch(&a, 0);
+     #if defined(__PPC__) && !defined(__PPC64__)
+       #error compiler built-in atomics are not supported on PowerPC 32-bit
+     #else
        return 0;
+     #endif
     ],
     [
 	AC_MSG_RESULT(yes)

